selfdrive/car/ford/carstate.py

```diff
@@ -36,6 +37,7 @@ def update(self, cp, cp_cam):
     # car speed
     ret.vEgoRaw = cp.vl["BrakeSysFeatures"]["Veh_V_ActlBrk"] * CV.KPH_TO_MS
     ret.vEgo, ret.aEgo = self.update_speed_kf(ret.vEgoRaw)
+    ret.vEgoCluster = 1.035 * ret.vEgo + CV.KPH_TO_MS
```

Comment:
is this on CAN anywhere?

Reply:
No, I checked all of the "kph", "mph" and "km/h" signals in the DBC and there are three which report the vehicle speed
- `Veh_V_ActlBrk` from ABS_ESC, I think this is probably from the wheel speed
- `Veh_V_ActlEng` from the PCM, I assume some transmission speed calculation
- `VehOverGnd_V_Est` from ABS_ESC, possibly filtered combination of both sources? It is clipped to > 1kph.
![image](https://github.com/commaai/openpilot/assets/4038174/2b2a4b53-f570-4445-8e0c-806cd0a24595)
Sometimes they are close and sometimes very different. But none of these are as high as the speed on the cluster.

Reply:
Maybe try also checking for unlabeled signals in each message with the cluster as the tx node in the DBC?

Reply:
I got some information, from somebody that does diving into the firmware/communication, that the digital cluster on F150s seems to use `Veh_V_ActlEng`.

Reply:
The cluster isn't mentioned in this DBC, I think because it's behind the gateway (on CAN "High Speed 3"), this DBC must just be for the HS2 bus. I looked at all of the messages in cabana for a route from my car and didn't find anything except lots of messages on bus 1, which I guess are OBD PIDs.

I found a new signal that looks like speed, but becaues it's on the OBD port it might not be worth looking into that further.
![image](https://github.com/commaai/openpilot/assets/4038174/538b4e18-a74d-44c8-a2e7-12f252407f47)
![image](https://github.com/commaai/openpilot/assets/4038174/5d190a46-7713-4eb4-9433-c8d23e9d6d1d)

Reply:
I'll check to see if `Veh_V_ActlEng` is a closer match

Reply:
Using `Veh_V_ActlEng` does seem to be a closer match, at least on the one test route I am using here
![image](https://github.com/commaai/openpilot/assets/4038174/4da0154f-864e-4192-b163-0bf84f512573)

Reply:
I think I broke the speed record today in an F-150, as per the cluster speed. 257mph while in reality it was about 70mph. 

The new sensor needs seems to be changing units across countries.